File: src/Listas3/pratica/Questao01.java
package Listas3.pratica;

import java.util.Scanner;

class Funcionario {

    String nomeFuncionario;
    int qtdeProducao;
    int horasTrabalhadas;

    public Funcionario(String nomeFuncionario, int qtdeProducao, int horasTrabalhadas) {
        this.nomeFuncionario = nomeFuncionario;
        this.qtdeProducao = qtdeProducao;
        this.horasTrabalhadas = horasTrabalhadas;
    }

    public int calcularProdutividade() throws ArithmeticException {
        return qtdeProducao / horasTrabalhadas;
    }
}

public class Questao01 {
    public static void main(String[] args) {
        Scanner entrada = new Scanner(System.in);

        try {
            String nome = entrada.nextLine();
            int producao = entrada.nextInt();
            int horas = entrada.nextInt();

            Funcionario funcionario = new Funcionario(nome, producao, horas);
            int produtividade = funcionario.calcularProdutividade();

            System.out.printf("Produtividade: %d peças por hora%n", produtividade);

        } catch (ArithmeticException e) {
            System.out.println("Erro: horas trabalhadas não podem ser zero");

        } catch (Exception e) {
            System.out.println("Erro inesperado");

        } finally {
            entrada.close();
        }
    }
}

File: Listas C++/Lista 3/questao01.cpp
#include <iostream>
#include <string>
#include <stdexcept>

class Funcionario {
    std::string nomeFuncionario;
    int qtdeProducao;
    int horasTrabalhadas;

public:
    Funcionario(std::string nomeFuncionario, int qtdeProducao, int horasTrabalhadas) {
        this->nomeFuncionario = nomeFuncionario;
        this->qtdeProducao = qtdeProducao;
        this->horasTrabalhadas = horasTrabalhadas;
    }

    int calcularProdutividade() {
        if (horasTrabalhadas == 0) {
            throw std::runtime_error("ArithmeticException");
        }
        return qtdeProducao / horasTrabalhadas;
    }
};

int main() {
    try {
        std::string nome;
        std::getline(std::cin, nome);

        int producao, horas;
        
        if (!(std::cin >> producao)) throw std::exception();
        if (!(std::cin >> horas)) throw std::exception();

        Funcionario funcionario(nome, producao, horas);
        int produtividade = funcionario.calcularProdutividade();

        std::cout << "Produtividade: " << produtividade << " peças por hora" << std::endl;

    } catch (const std::runtime_error& e) {
        std::cout << "Erro: horas trabalhadas não podem ser zero." << std::endl;
    } catch (const std::exception& e) {
        std::cout << "Erro inesperado" << std::endl;
    }

    return 0;
}